tools/statplot.py

```diff
@@ -237,50 +272,74 @@ def PlotData(self, action, type, xscale, yscale):
+                                    + info.get('statistic') + f'%{i}')
+                        elif 'components' in info.keys():
+                            for i in range(int(info.get('components'))):
+                                entries = numpy.append(
+                                    entries, info.get('name') + '%'
+                                    + info.get('statistic') + f'%{i}')
+                        elif 'material_phase' in info.keys():
+                            entries = numpy.append(
+                                entries, info.get('material_phase') + '%'
+                                + info.get('name') + '%'
+                                + info.get('statistic'))
+                        else:
+                            entries = numpy.append(
+                                entries, info.get('name') + '%'
+                                + info.get('statistic'))
+                    elif line.startswith('</header>'):
+                        break
+            return entries, num
 
-    def ReadData(self, statfile):
-        stats = []
-        for i, filename in enumerate(statfile):
-            failcount = 0
-            while failcount < 4:
-                try:
-                    stats.append(fluidity_tools.Stat(filename))
-                    break
-                except (TypeError, ValueError):
-                    time.sleep(0.2)
-                    failcount += 1
-            if failcount == 4:
-                stats.append(fluidity_tools.Stat(filename))
-        if len(stats) == 1:
-            return stats[0]
-        else:
-            return fluidity_tools.JoinStat(*stats)
+        if self.statfile[0].endswith('stat'):
+            entries, num = GatherEntries(self.statfile[0])
+            values = numpy.genfromtxt(self.statfile[0], skip_header=num + 1)
+            values = values.T[numpy.argsort(entries)].T
+            entries = entries[numpy.argsort(entries)]
+            if len(self.statfile) > 1:
+                for item in self.statfile[1:]:
+                    entriesTemp, num = GatherEntries(item)
+                    if numpy.array_equal(
+                            entries, entriesTemp[numpy.argsort(entriesTemp)]):
+                        valuesTemp = numpy.genfromtxt(item,
+                                                      skip_header=num + 1)
+                        valuesTemp = valuesTemp.T[numpy.argsort(entriesTemp)].T
+                        values = numpy.vstack((values, valuesTemp))
+                    else:
+                        sys.exit('Statfiles entries do not match')
+        elif self.statfile[0].endswith('detectors'):
+            entries = GatherEntries(self.statfile[0])[0]
+            values = numpy.fromfile(self.statfile[0] + '.dat',
+                                    dtype=numpy.float64)
+            ncols = entries.size
+            nrows = numpy.floor(values.size / ncols).astype(int)
```

Comment:
```suggestion
            nrows = values.size // ncols
```

Reply:
Just needed an additional int() as values.size can be a float, but good idea indeed.

Reply:
The size can be a float!?

Reply:
Woops, memory confusion, it is the quotient that can be a float, so indeed no need for int(). Thanks for paying attention.